exp/addons/controllers/clusterresourceset_controller.go

```diff
@@ -61,8 +61,9 @@ var (
 
 // ClusterResourceSetReconciler reconciles a ClusterResourceSet object
 type ClusterResourceSetReconciler struct {
-	Client  client.Client
-	Tracker *remote.ClusterCacheTracker
+	Client    client.Client
+	APIReader client.Reader
```

Comment:
If I got this right, the APIReader here serves the same goal of the LiveClient in https://github.com/kubernetes-sigs/cluster-api/pull/3899
If so, wondering if we can use the same name ...

Reply:
Yeah, I was wondering the same thing originally, but left it as `APIReader` here because it's only a reader and not a full client. Calling it `APIClient` feels a bit weird, but I'm all open for better naming.

Reply:
what about LiveReader?

Reply:
I'm fine with either. LiveReader does include more info (the fact that it's live).